server/src/main/java/org/opensearch/action/bulk/TransportBulkAction.java

```diff
@@ -766,6 +787,10 @@ void executeBulk(
         final AtomicArray<BulkItemResponse> responses,
         Map<String, IndexNotFoundException> indicesThatCannotBeCreated
     ) {
+        /*
+         * We are not wrapping the listener here to capture the response codes for performance benefits. It will
+         * be saving us an iteration over the responses array
+         */
```

Comment:
Favoring inline instrumentation instead of wrapping the listener.

```java
    void executeBulk(
        Task task,
        final BulkRequest bulkRequest,
        final long startTimeNanos,
        final ActionListener<BulkResponse> listener,
        final AtomicArray<BulkItemResponse> responses,
        Map<String, IndexNotFoundException> indicesThatCannotBeCreated
    ) {
        ActionListener<BulkResponse> docStatusAwareListener = ActionListener.wrap(response -> {
            for (BulkItemResponse ir : response) {
                if (ir != null) {
                    indicesService.incDocStatusCounter(ir.status());
                }
            }
            listener.onResponse(response);
        }, listener::onFailure);
        new BulkOperation(task, bulkRequest, docStatusAwareListener, responses, startTimeNanos, indicesThatCannotBeCreated).run();
    }
```